### PR TITLE
V8 Enable segments in Client-code

### PR DESCRIPTION
**Pull request by nielslyngsoe**
Draft PR, still in progress

### CODE REVIEW

File: src/Umbraco.Web.UI.Client/src/common/directives/components/content/umbvariantcontent.directive.js [hunk 1]
```diff
@@ -7,12 +7,11 @@
     var umbVariantContent = {
         templateUrl: 'views/components/content/umb-variant-content.html',
         bindings: {
-            content: "<",
+            content: "=",
             page: "<",
-            editor: "<",
+            editor: "=",
```
Comment (Shazwazza) [inline src/Umbraco.Web.UI.Client/src/common/directives/components/content/umbvariantcontent.directive.js]: Having 2 way binding goes against how components 'should' work. The standard approach to components is to have one way binding and then callbacks else we fall into the previous angular traps where you don't know what is actually happening. Any reason why we need this? https://docs.angularjs.org/guide/component#component-based-application-architecture

File: src/Umbraco.Web.UI.Client/src/common/directives/components/content/umbvariantcontenteditors.directive.js [hunk 2]
```diff
@@ -122,157 +118,61 @@
...
+                    segment: variantSegment
                 });
             }
             else {
                 //replace the editor for the same culture
                 vm.editors[index].content = variant;
             }
         }
-
-        function initVariant(variant, editorIndex) {
```
Comment (Shazwazza) [inline src/Umbraco.Web.UI.Client/src/common/directives/components/content/umbvariantcontenteditors.directive.js]: So where has this code gone? There are vast amounts of changes in this PR so i can't really follow along very well but I fear a lot of stuff that we discovered was necessary is now being undone. There's a ton of test cases that needs to be covered regarding split views, invariant properties, editing values and ensure they are syncing between split views, ensuring that the same model is not being used between split views except for invariant properties, etc...

File: src/Umbraco.Web.UI.Client/src/common/directives/components/content/umbvariantcontent.directive.js (hunk 1 above)
Comment (nielslyngsoe) [reply src/Umbraco.Web.UI.Client/src/common/directives/components/content/umbvariantcontent.directive.js]: Good input, let me have a look to see what we can do about that.

File: src/Umbraco.Web.UI.Client/src/common/directives/components/content/umbvariantcontenteditors.directive.js (hunk 2 above)
Comment (nielslyngsoe) [reply src/Umbraco.Web.UI.Client/src/common/directives/components/content/umbvariantcontenteditors.directive.js]: This method basically added copies of other parts of the data to the variant object, to enable sub-parts of the editor to use that data. But the cause of this hack we had to maintain that data from multiple places. Resulting in an overly complex code, that is hard to read and not very open for change.

That's basically why I have chosen to refactor this. Let's have a talk to go through changes, cause I like to get your view on the whole thing.

File: src/Umbraco.Web.UI.Client/src/common/directives/components/content/umbvariantcontent.directive.js (hunk 1 above)
Comment (nielslyngsoe) [reply src/Umbraco.Web.UI.Client/src/common/directives/components/content/umbvariantcontent.directive.js]: Gone now, I will mark this feedback as resolved.

**Comment by nielslyngsoe** [comment]
fixes https://github.com/umbraco/Umbraco-CMS/issues/7610

**Comment by nielslyngsoe** [comment]
fixes https://github.com/umbraco/Umbraco-CMS/issues/7611

**Comment by nielslyngsoe** [comment]
fixes https://github.com/umbraco/Umbraco-CMS/issues/7900

**Comment by nielslyngsoe** [comment]
Merged into this PR https://github.com/umbraco/Umbraco-CMS/pull/7973

**Pull request closed without merging**